client/lib/posts/utils.js

```diff
@@ -14,7 +15,13 @@ var postNormalizer = require( 'lib/post-normalizer' ),
 var utils = {
 
 	getEditURL: function( post, site ) {
-		return `/${post.type}/${site.slug}/${post.ID}`;
+		let basePath = '';
+
+		if ( ! includes( [ 'post', 'page' ], post.type ) ) {
+			basePath = '/edit';
+		}
+
+		return `${basePath}/${post.type}/${site.slug}/${post.ID}`;
```

Comment:
might consider `encodeURIComponent` on each part, just in case. Seems like they're all safe (except maybe `post.type`?) but it would future proof against changes. Not really critical.